### ISSUE TITLE
Poor error messages in report

### ISSUE THREAD

**Issue by hol430** [APSIMInitiative/ApsimX#4011]
If a user enters an invalid variable name into a report, apsim will silently ignore this variable name but will create an empty column for this variable in the database and fill the column with nulls. Conversely, if the user enters an invalid reporting frequency, we get an error but the error message is not very helpful. Would it be better to throw a fatal if an invalid reporting variable is found? That way at least the user will know they have a problem.

**Comment by hol430** [comment]
@lie112 I was considering modifying code in ApsimXException to include the name of the model in the error message. However I see that the clem code uses this exception class extensively and you seem to use some sort of syntax in the error messages...if I changed ApsimXException to automatically prepend something like 'Error in model leaf: ...', would that break your error messages?

**Comment by lie112** [comment]
I had used the format of the APSIM error message "ERROR IN FILE:" to identify parts of errors and format them with my own CLEM markup language. I use the MessageType and Message columns of the _Messages table. The problem was formatting the message which is only ever stored as a string.  Ideally the APSIMx error would contain a range of properties or provide a better formatted error message details (i.e. "|" separated or something) such that a UI display could include a clickable tree view of nested error messages in the future. I didn't want to fiddle with what was provided so worked around it. If we could come up with a error message format that would allow messages to be extracted I can work with that. We shouldn't get errors if you change the message details, CLEM will just stop reporting them properly in the formatted message window. See ApsimNG/Presenters/CLEM/MessagePresenter.cs to see how I pull out the various error message starting details that I had identified and determine what is a warning and what is an error before formatting the following error message.
